index.js

```diff
@@ -0,0 +1,121 @@
+window.onload = function () {
+    const canvas = document.getElementById("canvas");
+    const context = canvas.getContext("2d");
+    const BB = canvas.getBoundingClientRect();
+
+    canvas.width = window.innerWidth * 0.8;
+    canvas.height = window.innerHeight * 0.8;
+
+    const offsetX = BB.left;
+    const offsetY = BB.top;
+    const canvasWidth = canvas.width;
+    const canvasHeight = canvas.height;
+
+    let collidedList = [];
+    let nowDragging = null;
+    let rectangles = [];
+    let y = N;
+
+    for (let i = 0; i < rectanglesList.length; i++) {
+        rectangles.push(new Rectangle(200, y, rectanglesList[i].width, rectanglesList[i].height, rectanglesList[i].color));
+        y += rectanglesList[i].height + N;
+    }
+
+    canvas.onmousedown = mouseDown;
+    canvas.onmouseup = mouseUp;
+    canvas.onmousemove = mouseMove;
+
+    draw();
+
+    function clear() {
+        context.clearRect(0, 0, canvasWidth, canvasHeight);
+    }
+
+    function draw() {
```

Comment:
I think you should move as much functions as you can out of this scope, and pass any data that you might need through arguments. Only `context` and `rectangles` in this case. Or create a class out of this file instead

Reply:
Can I pass arguments to draw () if I use requestAnimationFrame?

Reply:
Hmm that would require some changes for sure, you'll probably need to cancel previous frames after updating the data. The best option would be to rewrite this file as a class, then you can better see what this file does in what order, and you can also reorganize variables that are used here

canvas, context, collidedList and nowDragging can be class fields
canvasWidth, canvasHeight, offsetX, offsetY can be calculated inside methods, there's no real need to store them (well, maybe it's worth it to cache offsets because sometimes they take a long time to calculate, but here it should be fine)
`y` is only used for creating rectangles, which could be moved with following forEach inside a separate method called something like `spawnRectangles`